File: server/src/main/java/au/com/codeka/warworlds/server/util/SmtpHelper.java
package au.com.codeka.warworlds.server.util;

import org.simplejavamail.email.Email;
import org.simplejavamail.mailer.Mailer;
import org.simplejavamail.mailer.config.ServerConfig;
import org.simplejavamail.mailer.config.TransportStrategy;

import java.util.Properties;

import au.com.codeka.warworlds.server.Configuration;

/**
 * Helper class for sending {@link Email} messages.
 */
public class SmtpHelper {
  public static final SmtpHelper i = new SmtpHelper();

  private Mailer mailer;

  public void start() {
    mailer = new Mailer(
        new ServerConfig(
            Configuration.i.getSmtp().getHost(),
            Configuration.i.getSmtp().getPort(),
            Configuration.i.getSmtp().getUserName(),
            Configuration.i.getSmtp().getPassword()),
        TransportStrategy.SMTP_TLS);

    Properties props = new Properties();
    props.put("mail.smtp.localhost", "codeka.com.au");
    mailer.applyProperties(props);
  }

  public void stop() {
  }

  public void send(Email email) {
    email.setFromAddress(
        Configuration.i.getSmtp().getSenderAddr(),
        Configuration.i.getSmtp().getSenderAddr());
    mailer.sendMail(email, true);
  }
}

File: server/src/main/java/au/com/codeka/warworlds/server/util/SmtpHelper.kt
package au.com.codeka.warworlds.server.util

import au.com.codeka.warworlds.server.Configuration
import org.simplejavamail.email.Email
import org.simplejavamail.mailer.Mailer
import org.simplejavamail.mailer.config.ServerConfig
import org.simplejavamail.mailer.config.TransportStrategy
import java.util.*

/**
 * Helper class for sending [Email] messages.
 */
class SmtpHelper {
  private var mailer: Mailer? = null
  fun start() {
    mailer = Mailer(
        ServerConfig(
            Configuration.i.smtp.host, Configuration.i.smtp.port, Configuration.i.smtp.userName,
            Configuration.i.smtp.password),
        TransportStrategy.SMTP_TLS)
    val props = Properties()
    props["mail.smtp.localhost"] = "codeka.com.au"
    mailer!!.applyProperties(props)
  }

  fun stop() {}
  fun send(email: Email) {
    email.setFromAddress(Configuration.i.smtp.senderAddr, Configuration.i.smtp.senderAddr)
    mailer!!.sendMail(email, true)
  }

  companion object {
    val i = SmtpHelper()
  }
}